polclassifier/interface/main.py

```diff
@@ -204,6 +251,57 @@ def train_evaluate_model_knn(split_ratio: float = 0.2, perform_search: bool = Fa
     return model, accuracy
 
 
+def load_speeches(min_word_count=400, sample_size=1000, parties_to_exclude=[], speeches_per_party = 20):
+
+    # Load data from feather file
+    raw_data_path = os.path.join(
+            LOCAL_PATH, "raw_data", "Corp_HouseOfCommons_V2.feather")
+    data = pd.read_feather(raw_data_path)
+
+    # Filter and clean data
+    data = clean_data(df=data, min_word_count=min_word_count, sample_size=sample_size, parties_to_exclude=parties_to_exclude)
+
+    # Split the data into training and testing sets
+    data_train, data_test = train_test_split(data, test_size=0.2, random_state=42, stratify=data["party"])
+
+    # Undersample data_test
+    grouped_data = data_test.groupby('party')
+    smaller_data_test = []
+    for party, group in grouped_data:
+        # Select randomly 20 speeches per party
+        sampled_group = group.sample(n=speeches_per_party, random_state=42)
+        # Add selected speeches to list
+        smaller_data_test.append(sampled_group)
+
+    df = pd.concat(smaller_data_test)
+
+    def tokenize_if_needed(text):
+        if isinstance(text, list):
+            return text  # Already tokenized
+        return text.split()  # Tokenize by splitting
+
+    # Apply the tokenization function to ensure all texts are tokenized
+    df['text'] = df['text'].apply(tokenize_if_needed)
+
+    def cut_middle_200(words):
+        if len(words) <= 200:
+            return " ".join(words)  # Return the original text if it's shorter than 200 words
+        else:
+            start_index = (len(words) // 2) - 100  # Calculate the start index of the middle 200 words
+            # Ensure the start index is non-negative
+            start_index = max(start_index, 0)
```

Comment:
It can stay but it would only be negative if the speech is shorter than 200 words, no? (and you already are taking care of in line 287